sdk/src/extras/disk.rs

```diff
@@ -389,60 +385,68 @@ pub mod types {
                 }
 
                 // If the chunk we just read is all zeroes, don't POST it.
-                if !chunk.iter().all(|x| *x == 0) {
-                    let encoded = base64::engine::general_purpose::STANDARD.encode(&chunk[0..n]);
-
-                    if senders[i % self.upload_thread_ct]
-                        .send((offset, encoded, n as u64))
+                let data = &buf[..n];
+                if !data.iter().all(|x| *x == 0) {
+                    // Failure to send indicates that all upload tasks exited early
+                    // due to errors on their end. We will return those errors below.
+                    if tx
+                        .send_async(Chunk {
+                            offset,
+                            data: data.to_vec(),
+                        })
                         .await
                         .is_err()
                     {
-                        // Failure to send indicates that the upload task exited early
-                        // due to an error on its end. We will return that error below.
                         break Ok(());
                     }
                 } else {
-                    // Bump the progress bar here to make it consistent
-                    self.upload_progress.fetch_add(n as u64, Ordering::Relaxed);
-                    self.progress_tx
-                        .send_replace(self.upload_progress.load(Ordering::Relaxed));
+                    // Bump the progress bar here to make it consistent.
```

Comment:
It took me a minute to remind myself what this meant; for your consideration--feel free to ignore
```suggestion
                    // Bump the progress bar here. We don't need to send this chunk so we've completed the relevant work.
```

Reply:
Good improvement, thanks.